frontend/docs/pages/home/durable-best-practices.mdx

```diff
@@ -0,0 +1,19 @@
+## Durable Execution Best Practices
+
+Durable tasks require a bit of extra work to ensure that they are not misused. An important concept in running a durable task is that the task should be **deterministic**. This means that the task should always perform the same sequence of operations in between retries.
+
+The deterministic nature of durable tasks is what allows Hatchet to replay the task from the last checkpoint. If a task is not deterministic, it may produce different results on each retry, which can lead to unexpected behavior.
+
+## Maintaining Determinism
+
+By following a few simple rules, you can ensure that your durable tasks are deterministic:
+
+1. **Only call methods available on the `DurableContext`**: a very common way to introduce non-determinism is to call methods within your application code which produces side effects. If you need to call a method in your application code which fetches data from a database, calls any sort of i/o operation, or otherwise interacts with other systems, you should spawn those tasks as a **child task** or **child workflow** using `RunChild`.
```

Comment:
note: in Python now, we don't have the `spawn` methods on the context. we could add those back as wrappers?